src/emulator/storage/index.ts

```diff
@@ -43,7 +36,7 @@ export class StorageEmulator implements EmulatorInstance {
   }
 
   get rules(): StorageRulesetInstance | undefined {
-    return this._rules;
+    return this._rulesManager!.ruleset;
```

Comment:
You could theoretically call this before calling `start`, where `this._rulesManager` is instantiated; similarly for `setRules` below. This is not a new problem: all `EmulatorInstance`s are like this. A couple ideas (not sure if within scope for this PR):

- Do the private constructor + async factory method combo I did for `StorageRulesManager`
- Make these things fail loudly if `start` has not been called

Thoughts?